java/org/apache/catalina/WebResourceSet.java

```diff
@@ -36,6 +37,10 @@
      */
     WebResource getResource(String path);
 
+    default WebResource getResource(String path, List<List<WebResourceSet>> allResources){
```

Comment:
This looks wrong. A single `WebResourceSet` shouldn't need to be passed a reference to `allResources`.